sale_product_approval/models/product_product.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2021 Open Source Integrators
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    sale_ok = fields.Boolean(
+        string="Can be added to Quote",
+        copy=False,
+        readonly=True,
+    )
+    sale_ok_confirm = fields.Boolean(
+        string="Can be Sold",
+        copy=False,
+        readonly=True,
+    )
+    candidate_sale = fields.Boolean(
+        string="Candidate to add to Quote",
+    )
+    candidate_sale_confirm = fields.Boolean(
+        string="Candidate to be Sold",
+    )
+    can_edit_candidate = fields.Boolean(compute="_compute_can_edit_candidate")
+
+    def _compute_can_edit_candidate(self):
+        for product in self:
+            product.can_edit_candidate = self.env.user.has_group(
+                "sale_product_approval.group_product_administrator"
+            )
+
+    @api.onchange("candidate_sale_confirm")
+    def _onchange_candidate_sale_confirm(self):
```

Comment:
Change the field `candidate_sale` as compute/readonly=False/stored field to avoid onchanges.